source/gui/settingsDialogs.py

```diff
@@ -773,14 +773,53 @@ def onSave(self):
 
 	def postSave(self):
 		if self.oldLanguage!=config.conf["general"]["language"]:
-			if gui.messageBox(
-				# Translators: The message displayed after NVDA interface language has been changed.
-				_("For the new language to take effect, the configuration must be saved and NVDA must be restarted. Press enter to save and restart NVDA, or cancel to manually save and exit at a later time."),
-				# Translators: The title of the dialog which appears when the user changed NVDA's interface language.
-				_("Language Configuration Change"),wx.OK|wx.CANCEL|wx.ICON_WARNING,self
-			)==wx.OK:
-				config.conf.save()
-				queueHandler.queueFunction(queueHandler.eventQueue,core.restart)
+			wx.CallAfter(gui.mainFrame._popupSettingsDialog, LanguageRestartDialog, self.oldLanguage)

```

Comment:
It concerns me a little that you're using _popupSettingsDialog here. Also the CallAfter doesn't seem strictly necessary, and I don't think that the dialog is modal now.

Now, this is a bit complex anyway, since there's no real standard for how to do this. I'm also struggling with this as part of #8006.

I propose.

1. Removing the self.Show() call from the dialog.
2. Change this into: `LanguageRestartDialog(self.oldLanguage).ShowModal()`